worlds/inscryption/Rules.py

```diff
@@ -0,0 +1,203 @@
+            "Act 3 - The Great Transcendence": self.has_transcendence_requirements,
+            "Act 3 - Boss Mycologists": self.has_mycologists_boss_requirements,
+            "Act 3 - Bone Lord Room": self.has_bone_lord_room_requirements,
+            "Act 3 - Luke's File Entry 1": self.has_battery_and_quill,
+            "Act 3 - Luke's File Entry 2": self.has_battery_and_quill,
+            "Act 3 - Luke's File Entry 3": self.has_battery_and_quill,
+            "Act 3 - Luke's File Entry 4": self.has_transcendence_requirements,
+            "Act 3 - Well": self.has_inspectometer_battery,
+            "Act 3 - Gems Drone": self.has_inspectometer_battery,
+            "Act 3 - Clock": self.has_gems_and_battery,  # Can be brute-forced, but the solution needs those items.
+        }
+        self.region_rules = {
+            "Act 2": self.has_act2_requirements,
+            "Act 3": self.has_act3_requirements,
+            "Epilogue": self.has_epilogue_requirements
+        }
+
+    def has_wardrobe_key(self, state: CollectionState) -> bool:
+        return state.has("Wardrobe Key", self.player)
+
+    def has_caged_wolf(self, state: CollectionState) -> bool:
+        return state.has("Caged Wolf Card", self.player)
+
+    def has_dagger(self, state: CollectionState) -> bool:
+        return state.has("Dagger", self.player)
+
+    def has_magnificus_eye(self, state: CollectionState) -> bool:
+        return state.has("Magnificus Eye", self.player)
+
+    def has_useful_act1_items(self, state: CollectionState) -> bool:
+        return state.has("Oil Painting's Clover Plant", self.player) and state.has("Squirrel Totem Head", self.player)
+
+    def has_all_epitaph_pieces(self, state: CollectionState) -> bool:
+        if self.world.options.epitaph_pieces_randomization.value == 0:
+            return state.has("Epitaph Piece", self.player, 9)
+        elif self.world.options.epitaph_pieces_randomization.value == 1:
+            return state.has("Epitaph Pieces", self.player, 3)
+        else:
+            return state.has("Epitaph Pieces", self.player, 1)
+
+    def has_camera_and_meat(self, state: CollectionState) -> bool:
+        return state.has("Camera Replica", self.player) and state.has("Pile Of Meat", self.player)
+
+    def has_monocle(self, state: CollectionState) -> bool:
+        return state.has("Monocle", self.player)
+
+    def has_obol(self, state: CollectionState) -> bool:
+        return state.has("Ancient Obol", self.player)
+
+    def has_epitaphs_and_forest_items(self, state: CollectionState) -> bool:
+        return self.has_camera_and_meat(state) and self.has_all_epitaph_pieces(state)
+
+    def has_act2_bridge_requirements(self, state: CollectionState) -> bool:
+        return self.has_camera_and_meat(state) or self.has_all_epitaph_pieces(state)
+
+    def has_tower_requirements(self, state: CollectionState) -> bool:
+        return self.has_act2_bridge_requirements(state) and self.has_monocle(state)
```

Comment:
Very minor optimization, since `has_monocle` checks for one item instead of varied requirements.
```suggestion
    def has_tower_requirements(self, state: CollectionState) -> bool:
        return self.has_monocle(state) and self.has_act2_bridge_requirements(state)
```